official/resnet/keras/keras_imagenet_main.py

```diff
@@ -166,11 +169,19 @@ def run(flags_obj):
       optimizer = tf.keras.mixed_precision.experimental.LossScaleOptimizer(
           optimizer, loss_scale=flags_core.get_loss_scale(flags_obj))
 
+    if flags_obj.enable_xla:
+      per_replica_batch_size = (
+          flags_obj.batch_size // strategy.num_replicas_in_sync)
```

Comment:
Maybe use `flags_obj.num_gpus`?

Reply:
strategy.num_replicas_in_sync is more general, especially this code may run on other devices in the future like TPU.

Reply:
It will break the tests with no distribution strategy (get_distribution_strategy can return None). Also we'll have to make a lot of other changes for this to work with other devices.

Reply:
Makes sense. `num_gpus` doesn't cover all the cases as well because you may run the CPU XLA program. I made a change to support non-XLA case as well, PTAL.